cts.json

```diff
@@ -2115,6 +2115,160 @@
       "result": [
         {"d": "f"}
       ]
+    },
+    {
+      "name": "filter, length function, string data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"a": "ab"}, {"a": "d"}],
+      "result": [
+        {"a": "ab"}
+      ]
+    },
+    {
+      "name": "filter, length function, array data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"a": [1,2,3]}, {"a": [1]}],
+      "result": [
+        {"a": [1,2,3]}
+      ]
+    },
+    {
+      "name": "filter, length function, missing data",
+      "selector" : "$[?length(@.a)>=2]",
+      "document" : [{"d": "f"}],
+      "result": []
+    },
+    {
+      "name": "filter, length function, non-array/string arg",
+      "selector" : "$[?length(1)>=2]",
+      "invalid_selector": true
```

Comment:
That's why `invalid_selector` is set to `true`.

Reply:
> Are you saying because `1` is a `Value` this is valid, and it should just return `Nothing`?

Yes.

> 
> If there was something like `$[?length(@.a)>=2]` where the data at `@.a` contained a `1`, then it should return `Nothing`, but an explicitly invalid argument seems like it shouldn't parse.

That would be a spec change which could be considered.

Reply:
Please see #368 for `length(1)`.